addon/application.go

```diff
@@ -223,57 +223,64 @@ func (h *AppFacts) Source(source string) {
 
 //
 // List facts.
-func (h *AppFacts) List() (list []api.Fact, err error) {
-	list = []api.Fact{}
-	path := Path(api.ApplicationFactsRoot).Inject(Params{api.ID: h.appId})
-	err = h.client.Get(path, &list, Param{Key: api.Source, Value: h.source})
+func (h *AppFacts) List() (facts api.FactMap, err error) {
+	facts = api.FactMap{}
+	key := api.FactKey("")
+	key.With(h.source, "")
```

Comment:
Looking the usage here, I wonder if key.Qualify(source) would be more useful then With(source, name)?
Would result in usages like:
```
key := api.FactKey("")
key.Qualify(h.source)
```
And 
```
key := api.FactKey(name)
key.With(h.source)
```